seed/challenges/08-coding-interview-questions-and-take-home-assignments/coding-interview-data-structure-questions.json

```diff
@@ -1051,49 +1120,87 @@
-        "    var previousNode;",
-        "    if(currentNode.element === element){",
-        "        head = currentNode.next;",
-        "    } else {",
-        "        while(currentNode.element !== element) {",
-        "            previousNode = currentNode;",
-        "            currentNode = currentNode.next;",
-        "        }",
+        "    var previousNode = this.head;",
+        "    var currentNode = this.head;",
         "",
+        "    while(currentNode) {",
+        "     if(currentNode.element === element) {",
+        "       if(currentNode === this.head) {",
+        "         this.head = this.head.next;",
+        "        }",
         "        previousNode.next = currentNode.next;",
+        "        this.length--;",
+        "    } else {",
+        "      previousNode = currentNode;",
+        "    }",
+        "      currentNode = currentNode.next;",
         "    }",
-        "",
-        "    length --;",
         "  };",
         "",
         "  // Only change code below this line",
         "",
         "  // Only change code above this line",
-        "};"
+        "}",
+        "",
+        "// Test your code",
+        "var conga = new LinkedList();",
+        "conga.add('Kitten');",
+        "conga.add('Puppy');",
+        "console.log(conga.indexOf('Puppy'))",
+        "console.log(conga.elementAt('0'))",
+        "conga.remove('Kitten');",
+        "console.log(conga.head;"
       ],
       "tests": [
         "assert((function(){var test = new LinkedList(); return (typeof test.indexOf === 'function')}()), 'message: Your <code>LinkedList</code> class should have a <code>indexOf</code> method.');",
         "assert((function(){var test = new LinkedList(); return (typeof test.elementAt === 'function')}()), 'message: Your <code>LinkedList</code> class should have a <code>elementAt</code> method.');",
-        "assert((function(){var test = new LinkedList(); test.add('cat'); test.add('dog'); test.add('kitten'); return test.size() === 3}()), 'message: Your <code>size</code> method should return the length of the linked list');",
+        "assert((function(){var test = new LinkedList(); test.add('cat'); test.add('dog'); test.add('kitten'); return test.length === 3}()), 'message: Your <code>size</code> method should return the length of the linked list');",
         "assert((function(){var test = new LinkedList(); test.add('cat'); test.add('dog'); test.add('kitten'); return test.indexOf('kitten') === 2}()), 'message: Your <code>indexOf</code> method should return the index of the given element.');",
         "assert((function(){var test = new LinkedList(); test.add('cat'); test.add('dog'); test.add('kitten'); return test.elementAt(1) === 'dog'}()), 'message: Your <code>elementAt</code> method should return at element at a given index.');"
       ],
       "solutions": [],
       "hints": [],
       "type": "waypoint",
       "challengeType": 1,
-      "translations": {}
+      "translations": {
```

Comment:
Please revert this line of change. We do not need to add empty translation blocks.